.github/workflows/rust.yml

```diff
@@ -20,3 +20,23 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+
+
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust
+        run: rustup update stable
+      - name: Install cargo-llm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Generate Code Coverage
+        run: cargo llvm-cov --all-features --workspace --lcov --output-path lcov.info
+      - name: Upload coverage to CodeCov
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: lcov.info
+          fail_ci_if_error: true
```

Comment:
`fail_ci_if_error` fails the build if upload fails? Seems like an unnecessary blocker in case the other service goes down.

Reply:
Yes true. Uploading to CodeCov should be just best effort and not a must to have.